source/globalCommands.py

```diff
@@ -1371,14 +1371,32 @@ def script_sayAll(self,gesture):
 	script_sayAll.category=SCRCAT_SYSTEMCARET
 
 	def _reportFormattingHelper(self, info, browseable=False):
-		formatConfig={
-			"detectFormatAfterCursor":False,
-			"reportFontName":True,"reportFontSize":True,"reportFontAttributes":True,"reportColor":True,"reportRevisions":False,"reportEmphasis":False,
-			"reportStyle":True,"reportAlignment":True,"reportSpellingErrors":True,
-			"reportPage":False,"reportLineNumber":False,"reportLineIndentation":True,"reportLineIndentationWithTones":False,"reportParagraphIndentation":True,"reportLineSpacing":True,"reportTables":False,
-			"reportLinks":False,"reportHeadings":False,"reportLists":False,
-			"reportBlockQuotes":False,"reportComments":False,
-			"reportBorderStyle":True,"reportBorderColor":True,
+		formatConfig = {
+			"detectFormatAfterCursor": False,
+			"reportFontName": True,
+			"reportFontSize": True,
+			"reportFontAttributes": True,
+			"reportSuperscriptsAndSubscripts": True,
+			"reportColor": True,
+			"reportRevisions": False,
+			"reportEmphasis": False,
+			"reportStyle": True,
+			"reportAlignment": True,
+			"reportSpellingErrors": True,
+			"reportPage": False,
+			"reportLineNumber": False,
+			"reportLineIndentation": True,
+			"reportLineIndentationWithTones": False,
+			"reportParagraphIndentation": True,
+			"reportLineSpacing": True,
+			"reportTables": False,
+			"reportLinks": False,
+			"reportHeadings": False,
+			"reportLists": False,
+			"reportBlockQuotes": False,
+			"reportComments": False,
+			"reportBorderStyle": True,
+			"reportBorderColor": True,

```

Comment:
I assume that these values are supposed to match the defaults from the config spec, is this true? It would be good to have some comments about intent on this list. Is there a reason the values aren't pulled directly from the config?

Reply:
It looks to me that the intent is to explicitly report all formatting-related info (regardless of user config) when it's explicitly requested.

Reply:
Does it matter that the following items (from config) are not included in this dict:

1. includeLayoutTables
1. reportTableHeaders
1. reportTableCellCoords
1. reportGroupings
1. reportLandmarks
1. reportArticles
1. reportFrames
1. reportClickable

Perhaps it would be better if this was a list of only the `True` items, then used as `if x in list`? Would that help to avoid this kind of error occurring again in the future?

Reply:
> Does it matter that the following items (from config) are not included in this dict:
> 
> 1. includeLayoutTables
> 2. reportTableHeaders
> 3. reportTableCellCoords
> 4. reportGroupings
> 5. reportLandmarks
> 6. reportArticles
> 7. reportFrames
> 8. reportClickable
> 
> Perhaps it would be better if this was a list of only the `True` items, then used as `if x in list`? Would that help to avoid this kind of error occurring again in the future?

This is expected, as those items aren't text formatting. Changing to a list differs from how the config spec works and probably too big of a change for a simple regression fix.

Reply:
That would also break backwards compatibility, as `getFormatFieldsSpeech`/`getFormatFieldsBraille` would now take a list (of `True` items) and not a dict.

Reply:
> This is expected, as those items aren't text formatting

Are you saying this because they aren't (yet) used by `getFormatFieldsSpeech` or `getFormatFieldsBraille`? 

Those items are all in the `documentFormatting` section of the config. Perhaps this helper should have a list of explicitly defined true items, and then enumerate the `documentFormatting` section adding all remaining items as false. EG something like:

``` python
# These are the options we want reported when reporting formatting manually.
# for full list of options that may be reported see the "documentFormatting" section of L{config.configSpec}
reportFormattingOptions = [ 
	"reportFontName",
	"reportFontSize",
	"reportFontAttributes",
	"reportSuperscriptsAndSubscripts",
	"reportColor",
	"reportStyle",
	"reportAlignment",
	"reportSpellingErrors",
	"reportLineIndentation",
	"reportParagraphIndentation",
	"reportLineSpacing",
	"reportBorderStyle",
	"reportBorderColor",
]

# Create a dictionary to replace the config section that would normally be 
# passed to getFormatFieldsSpeech / getFormatFieldsBraille
formatConfig  = dict()
from config import conf
for i in conf["documentFormatting"]:
	formatConfig [i] = i in reportFormattingOptions
```

Then we can be sure not to hit this limitation again. Perhaps also a comment in the `documentFormatting` to say "The command `reportFormatting` overrides the config see x"